### ISSUE TITLE
Responsibility to start and end game

### ISSUE THREAD

**Issue by JamesZhang2** [CornellDataScience/FiggieBot#16]
I think it is more reasonable for the game engine to start and end the game, rather than having an agent send requests for starting and ending the game. Once 4 or 5 players are added, we could start the game; once a certain number of rounds is reached (or if someone runs out of money), we could end the game.

Also, it would be nice if the number of seconds of a round is customizable (i.e. set as a constant in the constants.py file). This will facilitate debugging.

**Comment by ili23** [comment]
I agree with the second part about the time being customizable and we can look into implementing it soon.

I feel like the game should not automatically start or end since we are also building the backend out for live players and we want the players themselves to have control over starting and ending the game. However, I would like to hear your thoughts on this so let us know your reasoning for having the engine start the game once those conditions are met.

**Comment by JamesZhang2** [comment]
I agree. Should we then have a request signaling that an agent is ready? Also, Figgie can only be played with 4 to 5 players, so should we start the game once we have 4 or 5 players and everyone is ready? I don't think the agents should be responsible for sending the "start game" request. They just need to send the "ready" request and the backend will decide to start the game when everyone is ready.
